### DISCUSSION TITLE
failed to write request: Broken pipe (os error 32)

### DISCUSSION THREAD

**Discussion by AdamWang7535** [rwf2/Rocket#2557] [Questions] [answered]
proc macro `post` not expanded: failed to write request: Broken pipe (os error 32)
I'm using Arch linux and coding with Microsoft VS code

Kernel:
6.3.6-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 05 Jun 2023 15:12:57 +0000 x86_64 GNU/Linux

**Comment by SergioBenitez** [comment]
Can you post the complete error/terminal output? This is likely a `rustc` or `cargo` bug/issue. Try a `cargo clean` which may resolve things.

**Comment by AdamWang7535** [comment]
I installed an extension called rust-analyzer. When I run this program,everything is fine. But in VS code, there are always errors reported by the extension. It makes me confused, since I don't know whether I code correctly.
Here's the output of the extension:
proc macro `post` not expanded: failed to write request: Broken pipe (os error 32)rust-analyzer[unresolved-proc-macro](https://rust-analyzer.github.io/manual.html#unresolved-proc-macro)

**Reply by SergioBenitez** [reply, marked as answer]
Did you [enable `proc-macro` support in rust-analyzer](https://rust-analyzer.github.io/manual.html#configuration), as the link you posted indicates?

**Reply by AdamWang7535** [reply]
I disabled proc-macro just now and The problem was solved. Thank you for your help.
